### ISSUE TITLE
Could canvasXpress respect `theme(legend.justification = c(0, 1)` in ggplot2?

### ISSUE THREAD

**Issue by davidhodge931** [neuhausi/canvasXpress#149]
I want to be able to put the legend top-right, top-left or bottom-left as the alignment looks good

```
library(tidyverse)
library(palmerpenguins)
library(canvasXpress)

p <- penguins |> 
  ggplot() +
  geom_point(aes(x = flipper_length_mm, 
                 y = body_mass_g, 
                 colour = species)) +
  theme(legend.justification = c(0,1))

p

canvasXpress(p)
```
![image](https://github.com/neuhausi/canvasXpress/assets/55065152/1bb349ec-2483-4bc9-864b-0ae6e11e1734)

**Comment by neuhausi** [comment]
Fixed in 48.6
![cX-htmlwidget-f3a2c6d06713768fd836-cx](https://github.com/neuhausi/canvasXpress/assets/254923/3811530c-b311-44d1-aba0-ee6e864f8d53)

**Comment by davidhodge931** [comment]
Works awesome when legend is on right.

Could this work when the legend is on the top or bottom?

**Comment by neuhausi** [comment]
Yes. That will require a little bit of work since R flattens the legend when is top or bottom. Example?

**Comment by neuhausi** [comment]
I used an example in your website 
```
penguins |>
  gg_histogram(
    x = flipper_length_mm,
    col = species,
    title = "Penguin flipper length by species",
    subtitle = "Palmer Archipelago, Antarctica",
    caption = "Source: Gorman, 2020", 
    mode = light_mode_t(),
  ) +
  labs(colour = NULL, fill = NULL)
```
When I inspect the ggplot object the theme for the legend justification is ``` legend.justification" = c("0", "0.5")```
This is the plot I get with  CanvasXpress. Perhaps I don't understand the way the justification is set. Does this mean align the legend to 50% of the width? Could you help my understanding. I see your plot  that the legend is all the way to the left.

![cX-htmlwidget-5f0d625c494b7e33cfe2-cx](https://github.com/neuhausi/canvasXpress/assets/254923/f84df75b-3ed8-41e2-9329-be2768aaaf48)

**Comment by neuhausi** [comment]
I just released version 48.7. Please check and close issues as needed

**Comment by davidhodge931** [comment]
legend.justification element 1 refers to the hjust between 0 (left) and 1 (right).

legend.justification element 2 refers to the vjust between 0 (bottom) and 1 (top).

I use c(0, 1) for my `*_mode_r` theme with the legend on right, and c(0, 0.5) for the `*_mode_t` and `*_mode_b` one.

Cool, you are using ggblanket! Now you can see where my questions are coming from :)

I'll check and close all these once I am able to install 48.7. Not sure how to get this at the moment

**Comment by davidhodge931** [comment]
Also, there is `theme(legend.location = "plot")` or  `theme(legend.location = "panel")` to consider to. 

This says whether to consider the legend justification with regard to the plot or panel. 

I use panel for my `*_mode_r` themes and plot for my `*_mode_t` and `*_mode_b` themes

**Comment by neuhausi** [comment]
Ok So there are pretty much binary for the x coordinate yes or no?

Here are another two plots  I am fixing in the next release ;-)

![cX-htmlwidget-db2114ff29b7d3ae7020-cx](https://github.com/neuhausi/canvasXpress/assets/254923/0a88a471-f074-46b3-801b-68916e1c0494)

![cX-htmlwidget-fc621f62a3fc381792d5-cx](https://github.com/neuhausi/canvasXpress/assets/254923/1812ed2d-f30c-4093-af77-4cb3188abee5)

**Comment by neuhausi** [comment]
> legend.justification element 1 refers to the hjust between 0 (left) and 1 (right).
> 
> legend.justification element 2 refers to the vjust between 0 (bottom) and 1 (top).
> 
> I use c(0, 1) for my `*_mode_r` theme with the legend on right, and c(0, 0.5) for the `*_mode_t` and `*_mode_b` one.
> 
> Cool, you are using ggblanket! Now you can see where my questions are coming from :)

I want you to use CanvasXpress :-)

**Comment by davidhodge931** [comment]
They are any decimal between 0 and  1 (e.g. 0.86)

**Comment by neuhausi** [comment]
Fixed in version 48.9
![cX-htmlwidget-fc621f62a3fc381792d5-cx](https://github.com/neuhausi/canvasXpress/assets/254923/f4e438ae-d7b6-4df0-b496-47351a0905db)

**Comment by davidhodge931** [comment]
```
penguins |>
  gg_histogram(
    x = flipper_length_mm,
    col = species,
    title = "Penguin flipper length by species",
    subtitle = "Palmer Archipelago, Antarctica",
    caption = "Source: Gorman, 2020", 
  ) 
```

**Comment by davidhodge931** [comment]
Looks pretty good. Perhaps the alignment looks slightly too low?

![image](https://github.com/neuhausi/canvasXpress/assets/55065152/5d233e20-962e-4a50-aeb4-4f20632cbfdc)

**Comment by davidhodge931** [comment]
histogram is dropping the legend title 

![image](https://github.com/neuhausi/canvasXpress/assets/55065152/c228245f-f35f-4477-ae8b-d89fd8ece6ec)

**Comment by neuhausi** [comment]
Fixed in Version 49.1
